blockchain/source/contracts/Market.sol

```diff
@@ -679,10 +692,12 @@ contract Market is Ownable {
         return rate.mul(_price).mul(_period).div(1e18);
     }
 
-    function AddToBlacklist(uint dealID, bool blacklisted) internal {
-        if (msg.sender == deals[dealID].consumerID && blacklisted == true) {
+    function AddToBlacklist(uint dealID, BlacklistPerson role) internal {
+        require(msg.sender == deals[dealID].consumerID && uint64(role) <= 2 || uint64(role) == 0); // only consumer can blacklist
```

Comment:
Why not just to compare `role == BLACKLIST_NOBODY ` instead of `uint64(role) == 0)` here?